### PR TITLE
[Backtesting] allow custom exchange type

### PR DESCRIPTION
**Pull request by techfreaque**

### CODE REVIEW

File: Services/Interfaces/web_interface/models/backtesting.py [hunk 1]
```diff
@@ -150,6 +153,7 @@ def _start_backtesting(files, source, reset_tentacle_config=False, run_on_common
             else:
                 tentacles_setup_config = interfaces_util.get_bot_api().get_edited_tentacles_config()
             config = interfaces_util.get_edited_config()
+            config[commons_constants.CONFIG_EXCHANGE_TYPE] = exchange_type
```
Comment (GuillaumeDSM) [inline Services/Interfaces/web_interface/models/backtesting.py:156]: 👍

File: Trading/Mode/scripted_trading_mode/scripted_trading.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        super().__init__(config, exchange_manager)
+        self.producer = ScriptedTradingModeProducer
+        import backtesting_script
+        import profile_trading_script
+        self.register_script_module(profile_trading_script)
+        self.register_script_module(backtesting_script, live=False)
+
+
+class ScriptedTradingModeProducer(modes.AbstractScriptedTradingModeProducer):
+    pass
```
Comment (GuillaumeDSM) [inline Trading/Mode/scripted_trading_mode/scripted_trading.py:30]: we should not commit `Trading/Mode/scripted_trading_mode/` here

**Review by GuillaumeDSM — COMMENTED** [review]
To be merged after https://github.com/Drakkar-Software/OctoBot-Tentacles/pull/658

File: Services/Interfaces/web_interface/advanced_controllers/tentacles.py [hunk 3]
```diff
@@ -18,14 +18,33 @@
...
+    if dev_mode_is_on:
+        @advanced_controllers.advanced.route(tentacles_route)
+        @login.login_required_when_activated
+        @cross_origin(origins="*")
+        def tentacles():
+            return _tentacles()
+    else:
+        @advanced_controllers.advanced.route(tentacles_route)
+        @login.active_login_required
+        @cross_origin(origins="*")
```
Comment (Herklos) [inline Services/Interfaces/web_interface/advanced_controllers/tentacles.py:37]: Could you add a variable for the origin value please? So we can change it easily if needed

File: Services/Interfaces/web_interface/advanced_controllers/tentacles.py [hunk 4]
```diff
@@ -18,14 +18,33 @@
...
+from tentacles.Services.Interfaces.octo_ui2.models.octo_ui2 import import_cross_origin_if_enabled, dev_mode_is_on
 import tentacles.Services.Interfaces.web_interface.util as util
 import tentacles.Services.Interfaces.web_interface.models as models
 import octobot_commons.authentication as authentication
 
-
-@advanced_controllers.advanced.route("/tentacles")
-@login.active_login_required
-def tentacles():
+tentacles_route = "/tentacles"
```
Comment (Herklos) [inline Services/Interfaces/web_interface/advanced_controllers/tentacles.py:26]: Could you wrapp this code (the as tentacles-package route) in an annotation or a function to prévient duplicating code?

**Pull request closed without merging**